vms/platformvm/vm.go

```diff
@@ -274,6 +280,46 @@ func (vm *VM) Initialize(
 	return nil
 }
 
+func (vm *VM) periodicallyPruneMempool(frequency time.Duration) {
+	ticker := time.NewTicker(frequency)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-vm.onShutdownCtx.Done():
+			return
+		case <-ticker.C:
+			if err := vm.pruneMempool(); err != nil {
+				vm.ctx.Log.Debug("pruning mempool failed",
+					zap.Error(err),
+				)
+			}
+		}
+	}
+}
+
+func (vm *VM) pruneMempool() error {
+	vm.ctx.Lock.Lock()
+	defer vm.ctx.Lock.Unlock()
+
+	blockTxs, err := vm.Builder.PackBlockTxs(math.MaxInt)
```

Comment:
It seems like this approach could lead to some transactions invalidating others (based on their position in the "fake block"). I assume this is fine but we may want to comment?

Reply:
I think this is actually a benefit for pruning this way. Happy for a comment though. Seems related to aaron's comment about different verification paths